### PR TITLE
MM-30861: Fix racy test TestSaml2Login

### PR DESCRIPTION
**Pull request by agnivade**
https://mattermost.atlassian.net/browse/MM-30861

```release-note
NONE
```

### CODE REVIEW

File: web/saml.go [hunk 1]
```diff
@@ -118,14 +118,12 @@ func completeSaml(c *Context, w http.ResponseWriter, r *http.Request) {
 	switch action {
 	case model.OAUTH_ACTION_SIGNUP:
 		teamId := relayProps["team_id"]
-		if len(teamId) > 0 {
-			c.App.Srv().Go(func() {
```
Comment (agnivade) [inline web/saml.go:122]: @sbishel / @mkraft - Same question as earlier. I didn't see any particular cause for delays except for a few additional DB calls and websocket messages. Nothing that should cause any noticeable delay.

This is also achievable by deep copying the user and doing it in a goroutine. But I wanted to check if there was anything that I missed.

File: web/saml.go [hunk 2]
```diff
@@ -118,14 +118,12 @@ func completeSaml(c *Context, w http.ResponseWriter, r *http.Request) {
...
-				if err = c.App.AddUserToTeamByTeamId(teamId, user); err != nil {
-					mlog.Error(err.Error())
-				} else {
-					c.App.AddDirectChannels(teamId, user)
-				}
-			})
+		if teamId != "" {
+			if err = c.App.AddUserToTeamByTeamId(teamId, user); err != nil {
+				mlog.Error(err.Error())
```
Comment (streamer45) [inline web/saml.go:123]: We could just `break` here.

File: web/saml.go [hunk 3]
```diff
@@ -118,14 +118,12 @@ func completeSaml(c *Context, w http.ResponseWriter, r *http.Request) {
...
 		teamId := relayProps["team_id"]
-		if len(teamId) > 0 {
-			c.App.Srv().Go(func() {
-				if err = c.App.AddUserToTeamByTeamId(teamId, user); err != nil {
-					mlog.Error(err.Error())
-				} else {
-					c.App.AddDirectChannels(teamId, user)
-				}
-			})
+		if teamId != "" {
```
Comment (streamer45) [inline web/saml.go:121]: This could be done in the same line/scope.

**Review by streamer45 — APPROVED** [review]
Looks good :+1: 
Left a couple of minor syntaxing suggestions.

File: web/saml.go (hunk 1 above)
Comment (sbishel) [reply web/saml.go:122]: I'm not as familiar with this section, but tracing it through, I agree. Shouldn't cause noticable delay.

**Review by sbishel — APPROVED** [review]

**Review by mkraft — APPROVED** [review]

**Pull request merged**